lib/tasks/eu_exit_business_finder.rake

```diff
@@ -21,9 +21,40 @@ namespace :eu_exit_business_finder do
       updated_links = updated_links - EuExitFacetMigrationConfig::facet_values_to_remove
       updated_links = updated_links.uniq
       list.write_attribute('links', facet_values: { any: updated_links })
+      list.write_attribute('title', title_from_facets(updated_links))
       list.save
     end
   end
+
+  # Title is constructed from finder content item, so we have to retrieve the
+  # facet titles from the JSON using their content IDs.
+  def title_from_facets(chosen_facet_ids)
+    business_finder_json = Services.publishing_api.get_content('42ce66de-04f3-4192-bf31-8394538e0734')
+    all_facets = business_finder_json['links']['facet_group'][0]['links']['facets']
+    all_facet_ids_and_titles = flatten_facet_values_to_h(all_facets)
+    chosen_facet_ids_and_titles = override_special_titles(all_facet_ids_and_titles.filter { |content_id, _title|
```

Comment:
If you take my suggestion of doing the overrides in `flatten_facet_values_to_h` rather than having a separate method, a lot of this code could be removed.

`chosen_facet_ids_and_titles` wouldn't have to be created (i.e. remove lines 35 to 37).  You could go straight to `chosen_facet_titles`, e.g.
```
chosen_facet_titles = chosen_facet_ids.map { |chosen_facet_id| all_facet_ids_and_titles[chosen_facet_id] }
```